corgi/core/migrations/0058_change_software_build_primary_key.py

```diff
@@ -0,0 +1,116 @@
+                "PRIMARY KEY (uuid)",
+                # recreate the foreign key on softwarebuildtag
+                "ALTER TABLE core_softwarebuildtag DROP COLUMN tagged_model_id",
+                "ALTER TABLE core_softwarebuildtag ADD FOREIGN KEY (tagged_model_uuid) "
+                "REFERENCES core_softwarebuild (uuid) ON DELETE CASCADE",
+                # recreate the foreign key on component
+                "ALTER TABLE core_component DROP COLUMN software_build_id",
+                "ALTER TABLE core_component ADD FOREIGN KEY (software_build_uuid) "
+                "REFERENCES core_softwarebuild(uuid) ON DELETE CASCADE",
+            ],
+            reverse_sql=[
+                # add id columns to softwarebuildtag and populate with values
+                "ALTER TABLE core_softwarebuildtag ADD COLUMN tagged_model_id INTEGER",
+                "UPDATE core_softwarebuildtag "
+                "SET tagged_model_id=subquery.build_id "
+                "FROM (SELECT uuid, build_id"
+                "      FROM core_softwarebuild) AS subquery "
+                "WHERE core_softwarebuildtag.tagged_model_uuid = subquery.uuid",
+                # drop foreign key constraint on softwarebuildtag
+                "ALTER TABLE core_softwarebuildtag DROP CONSTRAINT "
+                "core_softwarebuildtag_tagged_model_uuid_fkey",
+                # Remove the old foreign key column as well
+                "ALTER TABLE core_softwarebuildtag DROP COLUMN tagged_model_uuid",
+                # add software_build_id column to component and populate withe values
+                "ALTER TABLE core_component ADD COLUMN software_build_id INTEGER",
+                "UPDATE core_component "
+                "SET software_build_id=subquery.build_id "
+                "FROM (select uuid, build_id"
+                "      FROM core_softwarebuild) AS subquery "
+                "WHERE core_component.software_build_uuid = subquery.uuid",
+                # drop foreign key constraint on component
+                "ALTER TABLE core_component DROP CONSTRAINT "
+                "core_component_software_build_uuid_fkey",
+                # remove the uuid field as well
+                "ALTER TABLE core_component DROP COLUMN software_build_uuid",
+                "ALTER TABLE core_softwarebuild DROP CONSTRAINT core_softwarebuild_pkey CASCADE",
+                "ALTER TABLE core_softwarebuild DROP COLUMN uuid",
+                "ALTER TABLE core_softwarebuild ADD PRIMARY KEY (build_id)",
+                "ALTER TABLE core_softwarebuildtag ADD FOREIGN KEY (tagged_model_id) "
+                "REFERENCES core_softwarebuild (build_id) ON DELETE CASCADE",
+                "ALTER TABLE core_component ADD FOREIGN KEY (software_build_id) "
+                "REFERENCES core_softwarebuild (build_id) ON DELETE CASCADE",
+            ],
+            state_operations=[
+                migrations.RemoveIndex(model_name="softwarebuild", name="core_softwarebuild_pkey"),
```

Comment:
Reviewing the documention on RunSQL and StateOperations again I don't think I need the RemoveIndex there. The index will be removed, and recreated by Postgres with the given SQL, but I don't think results in a state change according to Django.

Reply:
The reason for creating the SQL here is so that we can populate the foreign key relationships from softwarebuild_tag and component to the new softwarebuild primary key manually. Maybe I could have done this in Python too, but doing in SQL seems easier to me.

Reply:
Definitely a good idea to take a backup of stage before we merge this.  I tested the migration on the production data and it seems to work well.